### PR TITLE
RPC crate loan pallet

### LINKED ISSUE #273: add rpc crate for loans pallet and implement get_account_liquidity JSON-RPC
**Motivation**

We're going to open up the liquidation functionality to our community, so we would like to provide essential JSON-RPC interfaces for them to access the liquidation status easily.

The most important interface is `getAccountLiquidity`.

**Suggested Solution**

Add a rpc crate for loans pallet. such as https://github.com/open-web3-stack/open-runtime-module-library/tree/master/oracle/rpc

Then implement the JSON-RPC loans_getAccountLiquidity:

HTTP GET:
loans_getAccountLiquidity?account=5GBykvvrUz3vwTttgHzUEPdm7G1FND1reBfddQLdiaCbhoMd

Return:
```
{
  "address": "5GBykvvrUz3vwTttgHzUEPdm7G1FND1reBfddQLdiaCbhoMd",
  "liquidity": "72532",
  "shortfall": "0"
}
```

### PR DESCRIPTION
**Pull request by ayushmishra2005**
resolves #273

### CODE REVIEW

File: pallets/loans/rpc/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+name    = 'pallet-loans-rpc'
+version = '3.0.0'
+
+[dependencies]
+codec                           = { package = "parity-scale-codec", version = "2.0.0" }
+jsonrpc-core                    = "15.0.0"
+jsonrpc-core-client             = "15.0.0"
+jsonrpc-derive                  = "15.0.0"
+serde                           = { version = "1.0.124", features = ["derive"] }
+sp-runtime                      = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.5" }
```
Comment (kaichaosun) [inline pallets/loans/rpc/Cargo.toml:13]: branch seems outdated with other pallets.

File: pallets/loans/rpc/runtime-api/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+[package]
+authors = ['Parallel Team']
+edition = '2018'
+name    = 'pallet-loans-rpc-runtime-api'
+version = '3.0.0'
+
+[dependencies]
+codec           = { package = 'parity-scale-codec', version = '2.0.0', default-features = false, features = ['derive'] }
+sp-api          = { git = 'https://github.com/paritytech/substrate', branch = 'polkadot-v0.9.5', default-features = false }
```
Comment (kaichaosun) [inline pallets/loans/rpc/runtime-api/Cargo.toml:9]: Same.

File: pallets/loans/rpc/runtime-api/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#![cfg_attr(not(feature = "std"), no_std)]
```
Comment (kaichaosun) [inline pallets/loans/rpc/runtime-api/src/lib.rs:1]: Missing license header, also other new created files.

File: pallets/loans/rpc/runtime-api/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+use codec::Codec;
+use sp_runtime::{DispatchError, FixedU128};
+
+sp_api::decl_runtime_apis! {
+    pub trait LoansApi<AccountId> where
+        AccountId: Codec, {
+        fn get_account_liquidity(account: AccountId) -> Result<(FixedU128, FixedU128), DispatchError>;
```
Comment (kaichaosun) [inline pallets/loans/rpc/runtime-api/src/lib.rs:8]: Type alias for FixedU128 should be more readable, like 
`type Shortfalls = FixedU128`
`type Surplus = FixedU128`

Comment (kaichaosun) [reply pallets/loans/rpc/runtime-api/src/lib.rs:8]: Do we need an RPC to get all the shortfall accounts?

File: pallets/loans/rpc/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+[package]
+authors = ['Parallel Team']
+edition = '2018'
+name    = 'pallet-loans-rpc'
+version = '3.0.0'
+
+[dependencies]
+codec                           = { package = "parity-scale-codec", version = "2.0.0" }
```
Comment (GopherJ) [inline pallets/loans/rpc/Cargo.toml:8]: let's always use single quote in Cargo.toml

File: pallets/loans/rpc/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+use jsonrpc_derive::rpc;
+pub use pallet_loans_rpc_runtime_api::LoansApi as LoansRuntimeApi;
+use sp_api::ProvideRuntimeApi;
+use sp_blockchain::HeaderBackend;
+use sp_runtime::FixedU128;
+use sp_runtime::{generic::BlockId, traits::Block as BlockT};
+
+#[rpc]
+pub trait LoansApi<BlockHash, AccountId> {
+    #[rpc(name = "pallet_loans_get_account_liquidity")]
```
Comment (GopherJ) [inline pallets/loans/rpc/src/lib.rs:14]: `pallet` is not necessary https://github.com/open-web3-stack/open-runtime-module-library/blob/master/oracle/rpc/src/lib.rs#L15

File: pallets/loans/rpc/runtime-api/src/lib.rs (hunk 4 above)
Comment (ayushmishra2005) [reply pallets/loans/rpc/runtime-api/src/lib.rs]: This PR covers only `get_account_liquidity`

File: pallets/loans/rpc/src/lib.rs (hunk 6 above)
Comment (yz89) [inline pallets/loans/rpc/src/lib.rs]: This jsonrpc should be named `lonas_getAccountLiquidity`.

Comment (ayushmishra2005) [reply pallets/loans/rpc/src/lib.rs]: @yz89 Please check latest code

Comment (yz89) [reply pallets/loans/rpc/src/lib.rs]: Latest code is `loans_get_account_liquidity` not `loans_getAccountLiquidity`

Comment (ayushmishra2005) [reply pallets/loans/rpc/src/lib.rs]: Done

**Review by GopherJ — APPROVED** [review]

**Review by yz89 — APPROVED** [review]

**Pull request merged**